integration/clean/clean_ami.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+// +build clean
+
+package main
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	smithyTime "github.com/aws/smithy-go/time"
+	"log"
+	"time"
+)
+
+func main() {
+	err := cleanAMI()
+	if err != nil {
+		log.Fatalf("errors cleaning %v", err)
+	}
+}
+
+func cleanAMI() []error {
+	log.Printf("Begin to clean EC2 Instances")
+
+	cxt := context.Background()
+	defaultConfig, err := config.LoadDefaultConfig(cxt)
+	if err != nil {
+		return []error{err}
+	}
+	ec2client := ec2.NewFromConfig(defaultConfig)
+
+	// Get list of ami
+	nameFilter := types.Filter{Name: aws.String("name"), Values: []string{
+		"cloudwatch-agent-integration-test-ubuntu*",
+		"cloudwatch-agent-integration-test-al2*",
+		"cloudwatch-agent-integration-test-rhel8-base*",
+		"cloudwatch-agent-integration-test-rhel8-1*",
+		"cloudwatch-agent-integration-test-rhel8-2*",
+		"cloudwatch-agent-integration-test-rhel8-3*",
+		"cloudwatch-agent-integration-test-rhel8-4*",
+		"cloudwatch-agent-integration-test-ol8-1*",
+		"cloudwatch-agent-integration-test-ol8-2*",
+		"cloudwatch-agent-integration-test-ol8-3*",
+		"cloudwatch-agent-integration-test-ol8-4*",
+	}}
+
+	//get instances to delete
+	daysToKeep := 60
+	keepDuration := -1 * time.Hour * 24 * time.Duration(daysToKeep)
+	expirationDate := time.Now().UTC().Add(keepDuration)
+
+	describeImagesInput := ec2.DescribeImagesInput{Filters: []types.Filter{nameFilter}}
+	describeImagesOutput, err := ec2client.DescribeImages(cxt, &describeImagesInput)
+	if err != nil {
+		return []error{err}
+	}
+
+	var errors []error
+	for _, image := range describeImagesOutput.Images {
+		creationDate, err := smithyTime.ParseDateTime(*image.CreationDate)
```

Comment:
Why are we using an external library for parsing the datetime here? What's special about the format?

Reply:
This is because of the format of the *image.CreationDate. smithyTime is what I called smithyTime "github.com/aws/smithy-go/time". Which contains the conversion from aws string format to time struct. I think this looks cleaner than having to hardcode the format and this module is owned by aws so we should be good here.

Reply:
I just saw that this isn't necessarily external, but still confused about this. What requires the smithy-time library that isn't solved in the time stdlib?

Reply:
Noted. Yeah that's my bad for not seeing `/aws/` in the path of the import. I'm fine with this.